### PR TITLE
Floquet Notebooks

### PR DESCRIPTION
**Pull request by christian512**
This adds notebooks for the Floquet formalism provided in QuTiP. There are two notebooks: one for explaining the usage of the solver functions and one explaining their internal workings.

### CODE REVIEW

File: tutorials-v4/time-evolution/011_floquet_solver.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+---
+
+# Floquet Solvers
+
+Author: C. Staufenbiel, 2022
+
+### Introduction
+
+The *Floquet formalism* deals with periodic time-dependent systems. It is more efficient to use the Floquet approach to such problems than using the standard master equation solver `qutip.mesolve()`.
+
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/011_floquet_solver.md:22]: just a nitpick, i would say "can be more efficient" and "has a broader range of validity"?

File: tutorials-v4/time-evolution/011_floquet_solver.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                  args=args, Tsteps=1000)
+
+plot_expectation_values([result], ylabels=["<Z>"]);
+```
+
+### Floquet Markov Master Equation
+
+Similar to `mesolve()` we can also use the Floquet formalism to solve a master equation  for a dissipative quantum system. The corresponding function is `fmmesolve()`. However, the dissipation process is here described as a noise spectral-density function.
+
+For example we can define a a linear noise spectral-density as:
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/011_floquet_solver.md:82]: repeated 'a a' here

File: tutorials-v4/time-evolution/011_floquet_solver.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+# initial state
+psi0 = basis(2, 0)
+```
+
+### Floquet Schrödinger Equation 
+
+We can now use the `qutip.fsesolve()` function to solve the dynamics of the system using the Floquet formalism for the Schrödinger equation. The arguments are similar to the ones passed to `qutip.sesolve()`. There is an optional parameter `T` which defines the period of the time-dependence. If `T` is not given it is assumed that the passed `tlist` spans one period. Therefore we always pass `T` in this tutorial.
+
+The `Tsteps` argument to `fsesolve()` can be used to set the number of time steps in one period `T` for which the Floquet mods are precalculated. Increasing this number should result in a better numerical accuracy. `Tsteps` should be even!
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/011_floquet_solver.md:64]: mods = modes

File: tutorials-v4/time-evolution/012_floquet_formalism.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    [A / 2 * sigmax(), "sin({w}*t)".format(w=omega)],
+]
+```
+
+### Floquet modes and quasienergies
+For periodic Hamiltonians the solution to the Schrödinger equation can be represented by the Floquet modes $\phi_\alpha(t)$ and the quasienergies $\epsilon_\alpha$. We can obtain these for the initial time $t=0$ by using the function `floquet_modes()`. We can display for example the first floquet mode at $t=0$ using a Wigner distribution.
+
+```python
+f_modes0, f_energies = floquet_modes(H, T)
+plot_wigner(f_modes0[0]);
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/012_floquet_formalism.md:67]: this is an interesting use of plot_wigner!  can we discern anything from this?

File: tutorials-v4/time-evolution/012_floquet_formalism.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+```python
+plt.scatter(A_list / omega, quasienergies1, label="e1")
+plt.scatter(A_list / omega, quasienergies2, label="e2")
+plt.xlabel("A / w"), plt.ylabel("Quasienergies")
+plt.legend();
+```
+
+### Time evolution with floquet mode
+To calculate the time evolution of a random initial state $\psi(0)$, we have to decompose the state in the Floquet basis (formed by the floquet modes).
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/012_floquet_formalism.md:94]: super nitpick, sometimes Floquet is capitalized, sometimes not.  Maybe stick with capitalized everywhere?

File: tutorials-v4/time-evolution/012_floquet_formalism.md [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+
+```python
+# Decompose initial state
+psi0 = rand_ket(2)
+f_coeff = floquet_state_decomposition(f_modes0, f_energies, psi0)
+```
+
+The floquet mode $\phi_\alpha(t)$ for later times $t>0$ can be calculated using the wave function propagator $U(t,0)$ by:
+
+$$ \phi_\alpha(t) = exp(-i\epsilon_\alpha t / \hbar) \, U(t,0) \, \phi_\alpha(0) $$
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/012_floquet_formalism.md:108]: for me the latex spaces \,  are just rendering as commas....

File: tutorials-v4/time-evolution/012_floquet_formalism.md [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+plt.plot(tlist_period, p_ex_period, label="First period - precomputed")
+plt.legend(loc="upper right")
+plt.xlabel("Time"), plt.ylabel("Occupation prob.");
+```
+
+### Floquet Markov formalism
+
+We can also solve a master equation using the Floquet formalism. A detailed derivation of the Floquet-Markov formalism used here is given in [Grifoni et al., Physics Reports 304, 299 (1998)](https://www.sciencedirect.com/science/article/abs/pii/S0370157398000222) and in the [QuTiP docs](https://qutip.org/docs/latest/guide/dynamics/dynamics-floquet.html). Note that the functionality described here is summarised in the function `fmmesolve` described in the [floquet solver notebook](011_floquet_solver.md).
+
+The interaction with the bath is described by a noise spectrum, which does not include the temperature dependency. Hence, the definition is slightly different to the one in the Bloch-Redfield formalism.  For details see the derivation of the formalism.
```
Comment (nwlambert) [inline tutorials-v4/time-evolution/012_floquet_formalism.md:179]: Maybe can include a comment that the temperature can be set using w_th argument?

**Review by nwlambert — APPROVED** [review]

File: tutorials-v4/time-evolution/012_floquet_formalism.md (hunk 4 above)
Comment (christian512) [reply tutorials-v4/time-evolution/012_floquet_formalism.md:67]: I just wanted to show that the floquet mode is a typical quantum state. I guess for a more complicated Hamiltonian the Wigner plot would be more interesting.

**Pull request merged**